lib/keeper/controller.rb

```diff
@@ -7,22 +7,36 @@ def self.included(klass)
     end
 
     module InstanceMethods
+      JWT_REGEX = /[A-Za-z0-9\-_=]+\.[A-Za-z0-9\-_=]+\.?[A-Za-z0-9\-_=]+/
+
       def require_authentication
-        decoded_token = request_decoded_token
-        if decoded_token.nil?
-          not_authenticated
-        else
-          authenticated(decoded_token)
+        token = authentication_token
+        return not_authenticated if token.nil?
+
+        if token.version_mismatch? || token.pending?
+          new_claims = regenerate_claims(token)
+          token.rotate(new_claims)
+          self.authentication_token = token
         end
+
+        authenticated(token)
+      end
+
+      def regenerate_claims(_old_token)
```

Comment:
This is a function to be overloaded by the inheriting controller, if needed. `nil` is a default behavior. I guess inline documentation could be added.